### ISSUE TITLE
Insets and SafeArea

### ISSUE THREAD

**Issue by marazmone** [marazmone/ios-compose-showcase#7]
User Interface (UI) design is an essential aspect of mobile app development. It's the visual design that your users will interact with when using your app. Good UI design is crucial for mobile apps because it can significantly impact user engagement and satisfaction.

Here are a few reasons why UI design is essential in mobile development:

1. First impressions matter: The first thing users see when they open an app is the UI. If the UI is poorly designed and difficult to use, users will likely uninstall the app immediately. On the other hand, if the UI is well-designed, users are more likely to stay and engage with the app.
2. User Experience: A good UI design can create a seamless user experience. Mobile users expect fast, responsive, and intuitive interfaces that are easy to navigate. A well-designed UI can help users find what they're looking for quickly and efficiently, leading to a better user experience.
3. Branding and differentiation: UI design plays a crucial role in branding and differentiation. A well-designed UI can set your app apart from the competition and help establish your brand identity.
4. Retention: A well-designed UI can also help with user retention. Users are more likely to continue using an app that they find visually appealing and easy to use. Retention is essential in mobile app development because it's much easier and cheaper to retain existing users than to acquire new ones.

In conclusion, UI design is critical in mobile app development because it can significantly impact user engagement, satisfaction, and retention. A well-designed UI can create a seamless user experience, establish your brand identity, and set your app apart from the competition.

> This text was generated with the help GPT chat

|IOS|Android|
|---|---|
|<img src="https://user-images.githubusercontent.com/7988862/236704469-aca11b16-159c-4666-b8e2-c12e88ce85ec.png">|<img src="https://user-images.githubusercontent.com/7988862/236704470-7f4d6a25-1958-4184-9455-d90871366e50.png">|

UI with these effects is not suitable for the 2023 year.

**Need to change the application and work with an edge-to-edge design**

**Comment by marazmone** [comment]
Need to use the library - [insetsx](https://github.com/mori-atsushi/insetsx)

**Android:**

1. Create a `theme`
```
<?xml version="1.0" encoding="utf-8"?>
<resources>

    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
        <item name="android:statusBarColor">@android:color/transparent</item>
        <item name="android:navigationBarColor">@android:color/transparent</item>
        <item name="android:windowLightStatusBar">true</item>
    </style>
</resources>
```
2. Add a line to `MainActivity`
`WindowCompat.setDecorFitsSystemWindows(window, false)`

**IOS:**

Add `.ignoreSafeArea` with `.prefferedColorScheme` to `ComposeView`

```
struct ContentView: View {
    var body: some View {
        ComposeView()
                .ignoresSafeArea()
                .preferredColorScheme(.light)
    }
}
```

**Common:**

Add `contentPadding` to `LazyColumn` in `ListScreen`

**Comment by marazmone** [comment]
### Result

**IOS:**

| Start state | Middle state | End state |
|---|---|---|
|![IMG_0069](https://user-images.githubusercontent.com/7988862/236704180-350c76d9-531a-4a8d-a356-deede153c179.PNG)|![IMG_0070](https://user-images.githubusercontent.com/7988862/236704191-205d086a-2d6a-4a1b-838f-8dbc3123a3bc.PNG)|![IMG_0071](https://user-images.githubusercontent.com/7988862/236704205-916c6fb8-8db5-429e-8d4c-1aa55fcf6de1.PNG)|

**Android:**

| Start state | Middle state | End state |
|---|---|---|
|![photo_2023-05-07 23 54 07](https://user-images.githubusercontent.com/7988862/236704347-86d0e0e7-dedd-4013-87ec-4f2b3191debf.jpeg)|![photo_2023-05-07 23 54 05](https://user-images.githubusercontent.com/7988862/236704351-c475f13d-00cc-4311-b0b4-65c52ddc608c.jpeg)|![photo_2023-05-07 23 54 04](https://user-images.githubusercontent.com/7988862/236704354-82770d08-40dc-44ac-8c6c-956d4bb8edd7.jpeg)|

**It is perfect NOW!**
